### PR TITLE
[Utils] Installer:  Deprecate 0.12.x

### LINKED ISSUE #3432: feat: Deprecate 0.12.x and lower versions support in installer
### Summary

Due to the official release for `0.14.0`, and `0.13.0` for updating plug-ins for a lone time, we can deprecate `0.12.x` and lower for installer now.

### Details

Besides `0.12.x` and lower versions, we can remove the extensions as well, such as `image extension`, and `tensorflow-tools`.
I think this can simplfy the installer implementation much more.

### Appendix

_No response_

### PR DESCRIPTION
**Pull request by SAtacker**
Fixes #3432, #3199

- Ref- https://github.com/WasmEdge/WasmEdge/issues/3432

### CODE REVIEW

File: utils/install.py [hunk 1]
```diff
@@ -301,9 +301,8 @@ def compare(self, version2, separator=". |-", ignorecase=True):
 TENSORFLOW_LITE_P = "tensorflowlite"
 TENSORFLOW_DEPS = "tensorflow_deps"
 TENSORFLOW_LITE_DEPS = "tensorflow_lite_deps"
-TENSORFLOW_TOOLS = "tensorflow_tools"
 IMAGE = "image"
-EXTENSIONS = [TENSORFLOW, IMAGE]
+EXTENSIONS = [TENSORFLOW]
```
Comment (q82419) [inline utils/install.py:305]: You can remove all the extensions. We only reserve plugins.

File: utils/install.py [hunk 2]
```diff
@@ -317,20 +316,14 @@ def compare(self, version2, separator=". |-", ignorecase=True):
...
-    "Linux" + "arm64" + IMAGE: VersionString("0.10.0"),
-    "Linux" + "armv8" + TENSORFLOW: VersionString("0.10.0"),
-    "Linux" + "armv8" + IMAGE: VersionString("0.10.0"),
-    "Linux" + "aarch64" + TENSORFLOW: VersionString("0.10.0"),
-    "Linux" + "aarch64" + IMAGE: VersionString("0.10.0"),
-    "Darwin" + "x86_64" + TENSORFLOW: VersionString("0.10.0"),
-    "Darwin" + "x86_64" + IMAGE: VersionString("0.10.0"),
-    "Darwin" + "arm64" + TENSORFLOW: VersionString("0.10.0"),
-    "Darwin" + "arm" + TENSORFLOW: VersionString("0.10.0"),
+    "Linux" + "x86_64" + TENSORFLOW: VersionString("0.13.0"),
```
Comment (q82419) [inline utils/install.py:319]: Ditto.

**Review by q82419 — CHANGES_REQUESTED** [review]

**Comment by hydai** [comment]
@SAtacker Let's remove CentOS 8 Stream. It's EOL, so we are not able to install any packages from the package manager system.

**Comment by hydai** [comment]
Also, please update `python3 utils/install.py -v 0.13.5 --plugins wasi_nn-ggml-b2781` to `python3 utils/install.py -v 0.13.5 --plugins wasi_nn-ggml-b3075`.

File: utils/install.py [hunk 3]
```diff
@@ -707,129 +677,33 @@ def is_default_path(args):
...
-
-
 def install_tensorflow_extension(
     args,
     compat,
     download_tf_=False,
     download_tf_lite_=False,
     download_tf_deps_=False,
     download_tf_lite_deps_=False,
-    download_tf_tools_=False,
```
Comment (q82419) [inline utils/install.py:809]: I think the `download_tf_` and `download_tf_lite_` can be removed, too?

File: utils/install.py [hunk 4]
```diff
@@ -707,129 +677,33 @@ def is_default_path(args):
...
 ):
     global CONST_release_pkg, CONST_lib_ext, CONST_lib_dir, CONST_env_path
 
     download_tf = download_tf_
     download_tf_lite = download_tf_lite_
     download_tf_deps = download_tf_deps_
     download_tf_lite_deps = download_tf_lite_deps_
-    download_tf_tools = download_tf_tools_
```
Comment (q82419) [inline utils/install.py:817]: Ditto.

File: utils/install.py [hunk 5]
```diff
@@ -707,129 +677,33 @@ def is_default_path(args):
...
 
     logging.debug(
-        "install_tensorflow_extension: %s %s %s %s %s",
+        "install_tensorflow_extension: %s %s %s %s",
         download_tf,
         download_tf_lite,
         download_tf_deps,
         download_tf_lite_deps,
-        download_tf_tools,
```
Comment (q82419) [inline utils/install.py:825]: Ditto.

**Review by q82419 — CHANGES_REQUESTED** [review]
You can remove the download link from `"second-state/WasmEdge-tensorflow"` because it provides the extensions. Reserve the `"-deps"` only.

And also, `tf_deps_version` is fixed as `"TF-2.12.0-CC"` now because we use this version after `0.13.0`. And beware that from this version, `aarch64` is supported for the deps.

**Comment by hydai** [comment]
Hi @SAtacker 
Thanks for this PR. I found that `test-python-install-script / manylinux2014 aarch64 (pull_request)` failed due to some weird issue. This error should not related to your changes here. Maybe we can move forward to merge this PR first and figure out the above error later. WDYT?

**Comment by SAtacker** [comment]
> Hi @SAtacker Thanks for this PR. I found that `test-python-install-script / manylinux2014 aarch64 (pull_request)` failed due to some weird issue. This error should not related to your changes here. Maybe we can move forward to merge this PR first and figure out the above error later. WDYT?

I see. Let's do it!

**Review by hydai — APPROVED** [review]
@q82419 PTAL

File: utils/install.py [hunk 6]
```diff
@@ -1340,24 +1075,13 @@ def set_consts(args, compat):
...
     CONST_urls = {
         WASMEDGE: "https://github.com/WasmEdge/WasmEdge/releases/download/{0}/WasmEdge-{0}-{1}".format(
             args.version, compat.release_package_wasmedge
         ),
         WASMEDGE_UNINSTALLER: "https://raw.githubusercontent.com/WasmEdge/WasmEdge/{0}/utils/uninstall.sh".format(
             args.uninstall_script_tag
         ),
-        IMAGE: "https://github.com/second-state/WasmEdge-image/releases/download/{0}/WasmEdge-image-{0}-{1}".format(
-            args.image_version, local_release_package_im
-        ),
```
Comment (q82419) [inline utils/install.py:1360]: Delete the `TENSORFLOW` and `TENSORFLOW_LITE` entries.

**Review by q82419 — CHANGES_REQUESTED** [review]

**Review by hydai — APPROVED** [review]
LGTM. @q82419 PTAL

**Review by q82419 — APPROVED** [review]
LGTM

**Pull request merged**